gen_data/gen_test_m24_single.py

```diff
@@ -0,0 +1,475 @@
+            
+            if make_novel_all_one_class:
+                # Make all unknown activities single class
+                assert 'novel' not in known_classes
+                activity = 'novel'
+
+            # TODO: decide what to do with red_light when at novelty_timestamp
+            # but sampling from known
+            if red_light is None:
+                red_light = vid
+                red_light_det = n
+        else:
+            raise ValueError(f'Unkown novelty_str: {novelty_str}')
+
+        df.loc[len(df.index)] = [
+            vid,
+            is_novel,
+            1,
+            activity,
+            is_novel & is_spatial,
+            is_novel & is_temporal,
+        ]
+
+        del combine_known_unknown[vid_id]
+
+    assert len(combine_known_unknown) == 0
+  
+    metadata = {
+        "protocol": protocol,
+        "num_total_samples": num_total_samples,
+        "round_size": round_size,
+        "difficulty": None,  # figure out what this should be
+        "distribution": None,  # figure out what this should be
+        "n_rounds": math.ceil(
+            num_total_samples / round_size
+        ),  # TODO: confirm that this should be ceil
+        "representation": None,  # figure out what this should be
+        "threshold": 0.5,  # TODO: don't hardcode this
+        "pre_novelty_batches": math.ceil(
+            red_light_det / round_size
+        ),  # TODO: double check if this should be ceil or floor
+        "feedback_max_ids": math.ceil(0.1 * round_size),  # don't hardcode this
+        "known_classes": len(known_classes),
+        "novel_classes": len(unknown_classes),
+        "red_light": red_light,  # see TODO comment above
+        "detection": red_light_det,
+        "degree": 1,  # shouldn't be hardcoded
+        "prob_novel": prob_novel_class,  # check not  'prop_novel'
+        "novel_type": aug_type,
+        "seed": seed,
+        "actual_novel_activities": unknown_classes,  # maybe wrong
+        "max_novel_classes": len(
+            set(unknown_videos_dict.values())
+        ),  # not sure what this means, likely wrong value
+    }
+
+    return df, metadata
+
+
+@click.command(help="Generates multiple SAIL-on test runs.")
+@click.option(
+    "--training_file_csv",
+    "-k",
+    "training_file_csv",
+    default=None,
+    help="Filepath to a csv with m24 novelty types that was trained on",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
+@click.option(
+    "--testing_file_csv",
+    "-u",
+    "testing_file_csv",
+    default=None,
+    help="Filepath to a csv with m24 novelty types for testing",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False),
+)
+@click.option("--num_groups", default=1, type=int,
+    help = "# of groups to generate tests for")
+@click.option("--num_runs", default=1, type=int,
+    help = "# of runs for each group")
+@click.option("--num_samples_per_run", default=10, type=int,
```

Comment:
Should the default be set to 1024, since that is the standard length of the test?